Added statistics to review results for questions

rmkeezer opened the pull request:
This adds a d3 chart to statistics for viewing question results

## Changes <!-- What does this pull request do/change? If this is a WIP, add some checkboxes! -->
Adds controller an d3 chart to statistics route

## Testing <!-- What's the best way to try out the changes manually? -->
got to statistics if there are answers to questions

## Pre-merge checklist <!-- Things that should be done before merging. Check and say N/A if an item is not applicable. -->

- [ ] Build succeeds locally and in CI
- [x] Proper documentation
- [x] Tests
- [x] Git history clean <!-- Mistake commits and unnecessary merge commits rebased away, proper commit message conventions followed: http://chris.beams.io/posts/git-commit/ https://seesparkbox.com/foundry/atomic_commits_with_git -->

rmkeezer committed:
Added stats controller and template for d3 viz

rmkeezer committed:
Merge branch 'master' into stats

rmkeezer committed:
Completed stacked bar chart for questions

rmkeezer committed:
Merge branch 'master' into stats

strburst commented on app/gulp/sass/app.scss:
```diff
@@ -3,3 +3,25 @@
+  border-color: black;
+  pointer-events: none;
+  opacity: 0.3;
+}
```
Is this `div.tooltop` on every page or d3 graph? If not, I would recommend creating a file like `_statistics.scss` and `@import`ing it instead.

strburst commented on app/gulp/js/statistics.js:
```diff
@@ -0,0 +1,133 @@
+    draw(data);
+  }
+  showStats();
+});
```
Remove all the unused variables and `console.log`s that the linter is complaining about (they should be unnecessary anyway).

I'm also seeing a TypeError.
```
Uncaught TypeError: Cannot read property 'length' of undefined
    at statistics.js:108
    at statistics.js:129
    at HTMLDocument.<anonymous> (jquery.turbolinks.self-176b981….js?body=1:30)
    at HTMLDocument.dispatch (jquery.self-bd7ddd3….js?body=1:5227)
    at HTMLDocument.elemData.handle (jquery.self-bd7ddd3….js?body=1:4879)
    at Object.trigger (jquery.self-bd7ddd3….js?body=1:5131)
    at HTMLDocument.<anonymous> (jquery.self-bd7ddd3….js?body=1:5861)
    at Function.each (jquery.self-bd7ddd3….js?body=1:371)
    at jQuery.fn.init.each (jquery.self-bd7ddd3….js?body=1:138)
    at jQuery.fn.init.trigger (jquery.self-bd7ddd3….js?body=1:5860)
    at HTMLDocument.onLoad (jquery.turbolinks.self-176b981….js?body=1:35)
    at fire (jquery.self-bd7ddd3….js?body=1:3233)
    at Object.fireWith [as resolveWith] (jquery.self-bd7ddd3….js?body=1:3363)
    at Function.ready (jquery.self-bd7ddd3….js?body=1:3583)
    at HTMLDocument.completed (jquery.self-bd7ddd3….js?body=1:3618)
```

strburst commented on app/views/layouts/application.html.haml:
```diff
@@ -51,7 +51,7 @@
               Settings
         %hr
-    #content.container
+    %div#content.container{ :style => "width:auto; height:100%" }
```
`%div` is the default tag, so it's better to leave it off. Again, put this inline style in an actual CSS class.

strburst commented on app/gulp/sass/app.scss:
```diff
@@ -3,3 +3,25 @@
+body {
+  height: calc(100% - 60px);
+  width: 100%;
+}
```
This CSS is adding a tiny scrollbar to every page for some reason. Also, maybe we should avoid using `calc` if it's possible (I checked caniuse and 15% of people are using browsers that don't have it). I would also add layout-related CSS that goes on every page into `_layout.scss`.

strburst commented on db/schema.rb:
```diff
@@ -1,84 +0,0 @@
-  add_foreign_key "lecturetaggings", "questions"
-  add_foreign_key "topictaggings", "questions"
-  add_foreign_key "topictaggings", "topics"
-end
```
It looks like you've deleted `schema.rb`...that doesn't look right. I think this is causing the CI build to fail.

strburst commented on app/views/statistics/index.html.haml:
```diff
@@ -0,0 +1,5 @@
+
+.stats_class{ data: {results: @results.to_json, questions: @questions.to_json } }
+
+%svg#graph{ :style => "height:100%; width:100%;" }
```
Put this inline style in an actual CSS class.

strburst reviewed (changes requested):
It would be nice if the graph only had tick marks at integers.

rmkeezer committed:
Merge branch 'master' into stats

rmkeezer committed:
Fixed labels for stacked chart

rmkeezer committed:
Fixed unnessary scrollbar

rmkeezer committed:
Added data for user graph

rmkeezer committed:
Added yscale to 1000 integers

strburst committed:
Style cleanup and remove unused data

Stop computing and sending the 'users' and 'uresults' properties in the
statistics page, since they weren't used by the client code.

strburst committed:
Disable failing statistics test for now

strburst committed:
Refactor styles

This moves some html/body rules to _layout.scss and gives the main
statistics graph a proper name.

strburst reviewed (approved):
@fvosburgh said he would look into the test failure, but I'm merging this for now.
